keras/layers/core.py

```diff
@@ -16,21 +16,65 @@
 srng = RandomStreams(seed=np.random.randint(10e6))
 
 class Layer(object):
-    def __init__(self):
+    def __init__(self, name, prev, input_dim):
         self.params = []
+        self.name = name
+        self.prev = prev
+
+        if type(self.prev) is str:
+            self.prev = [self.prev] # single string or a list of strings under one op
+
+        self.input_dim = input_dim
+        self.output_dim = (None, )
+
+    def setup(self):
+        '''
+            Contains code that depends on values that are optional but can be understood from context (ex: input_dim)
+        '''
+        return
+
+    def connect(self, node):
+        self.previous = node
```

Comment:
>self.prev is the name of the previous layer(as a string)
>self.previous is the actual previous layer object 

Maybe worth commenting boldly and/or naming them more distinctly? Or, perhaps, could the two just be unified? Why can't a user pass a layer object directly in place of a name?

>As a standard, all dimensions are passed around as tuples. It can be in any combination, but the calculations should hold good through the layers.

I'm not sure if that's answering the question about `input_dim`; if so, then I'm still confused about what it means for `input_dim` to be `None` and why `connect` only checks for `(None,)` as opposed to e.g. `(None, None)`.